### PR TITLE
Use placeholder image for broken images

### PR DESCRIPTION
**Pull request by wulman16**
When an image does not resolve successfully, Thumbs Up will
now use a placeholder image with the requested dimensions instead.

404 errors are changed to 400 errors to better reflect the nature
of potential issues with malformed image requests.

The `sharp` package is updated so that the project can be installed
locally. Vulnerabilities in outdated dependencies are resolved.

The README is updated with current instructions on how to install
and test the project locally.

### CODE REVIEW

**Comment by bmckeough-showcaseidx** [comment]
@wulman16 I just started looking at this PR. I get a "Cannot find module '../build/Release/sharp.node'" error when I try to follow the README directions. Do you have any insight into why I would be getting this error?

```
bmckeough@Brians-MBP thumbs-up % git branch   
  cloud-function
* local-config
  master
  staging
  update_readme_gcp_cf
  willulman-ch20357-improve-how-our-image-service-thumbs-up-handles
bmckeough@Brians-MBP thumbs-up % nvm version
v8.17.0
bmckeough@Brians-MBP thumbs-up % npm install
removed 3 packages and audited 184 packages in 1.006s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

bmckeough@Brians-MBP thumbs-up % node index.js
module.js:550
    throw err;
    ^

Error: Cannot find module '../build/Release/sharp.node'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/bmckeough/Projects/thumbs-up/node_modules/sharp/lib/constructor.js:10:15)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
```

**Comment by wulman16** [comment]
@bmckeough-showcaseidx I would try deleting the `node_modules` directory and then running `npm install` again.

File: src/server.js [hunk 1]
```diff
@@ -23,8 +24,8 @@ module.exports = (req, res) => {
   if (isDomain(domain)) {
     return fetch(res, server, path, query)
       .then(sendImage(req, res, dimensions))
-      .catch(sendError(res))
+      .catch(() => sendPlaceholder(res, dimensions))
```
Comment (bmckeough-showcaseidx) [inline src/server.js:27]: The catch and call to sendPlaceholder here seems somewhat duplicative to the catch and call to sendPlaceholder in sendImage. Can you clarify for me why this is needed in both places? Is there not a way to make any exception encountered by sendImage bubble up to this catch?

**Review by bmckeough-showcaseidx — COMMENTED** [review]
@wulman16 Thanks for the PR on thumbs-up. I had a couple of questions. Can you educate me?

File: src/server.js (hunk 1 above)
Comment (wulman16) [reply src/server.js:27]: After working on this for about an hour, I don't think there is a way to bubble up this error as these promises are currently written. I think you were right in that these promises are constructed in a non-standard way, so as it stands now we will need the catch and call to `sendPlaceholder()` in both places. Unfortunately, my understanding of promises isn't good enough to be able to explain why.

**Pull request merged**